src/modules/app/route-configs/EventsRouteConfig.js

```diff
@@ -52,16 +54,20 @@ class EventsRouteConfig implements RouteConfig<EventsRouteParamsType, RequiredPa
     return this.getRoutePath({ city, language })
   }
 
-  getRequiredPayloads = (payloads: AllPayloadsType): RequiredPayloadsType => ({ events: payloads.eventsPayload })
+  getRequiredPayloads = (payloads: AllPayloadsType): RequiredPayloadsType => ({
+    cities: payloads.citiesPayload,
+    events: payloads.eventsPayload
+  })
 
-  getPageTitle = ({ t, payloads, cityName, location }) => {
-    if (!cityName) {
+  getPageTitle = ({ t, payloads, location }) => {
+    const city = payloads.cities.data && payloads.cities.data.find(_city => _city.code === location.payload.city)
```

Comment:
I really hate that we have to do this checks all the time...

Reply:
Maybe we could refactor this by adding the current city model to the location or something like that...

Reply:
Of course not in the scope of this PR

Reply:
Yeah, we could think about replacing `cityName: ?string` with `cityModel: ?CityModel` in `getPageTitle`